config/v1/techpreview.dns.testsuite.yaml

```
@@ -0,0 +1,89 @@
+apiVersion: apiextensions.k8s.io/v1 # Hack because controller-gen complains if we don't have this
+name: "[Stable] DNS"
+crd: 0000_10_config-operator_01_dns-TechPreviewNoUpgrade.crd.yaml
+tests:
+  onCreate:
+  - name: Should be able to create a minimal DNS
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: DNS
+      spec: {} # No spec is required for a DNS
+    expected: |
+      apiVersion: config.openshift.io/v1
+      kind: DNS
+      spec: {}
+  - name: Should be able to specify an AWS role ARN for a private hosted zone
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: DNS
+      metadata:
+        creationTimestamp: null
+        name: cluster
+      spec:
+        platform:
+          type: AWS
+          aws:
+            privateZoneIAMRole: arn:aws:iam:123456789012:role/foo
+    expected: |
+      apiVersion: config.openshift.io/v1
+      kind: DNS
+      metadata:
+        creationTimestamp: null
+        name: cluster
+      spec:
+        platform:
+          type: AWS
+          aws:
+            privateZoneIAMRole: arn:aws:iam:123456789012:role/foo
+  - name: Should not be able to specify unsupported platform
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: DNS
+      metadata:
+        creationTimestamp: null
+        name: cluster
+      spec:
+        platform:
+          type: Azure
+          azure:
+            privateZoneIAMRole: arn:aws:iam:123456789012:role/foo
+    expectedError: "Invalid value: \"string\": allowed values are '' and 'AWS'"
+  - name: Should not be able to specify invalid arn
```

*nit* I know I suggested this, but probably better to be more specific for the test name for debug-ablity.
```suggestion
  - name: Should not be able to specify invalid AWS role ARN
```